py3status/modules/online_status.py

```diff
@@ -57,35 +58,56 @@ class Meta:
             ],
         }
 
-    def _connection_present(self):
-        if '://' in self.url:
-            try:
-                urlopen(self.url, timeout=self.timeout)
-            except:
-                return False
-            else:
-                return True
-        else:
-            fnull = open(os.devnull, 'w')
-            return subprocess.call(['ping', '-c', '1', self.url],
-                                   stdout=fnull, stderr=fnull) == 0
+    def post_config_hook(self):
+        self.color_on = self.py3.COLOR_ON or self.py3.COLOR_GOOD
+        self.color_off = self.py3.COLOR_OFF or self.py3.COLOR_BAD
+        self.use_ping = None
+
+    def _ping(self):
+        try:
+            if '://' in self.url:
+                ping_url = urlsplit(self.url).netloc
+            self.py3.command_output(['ping', '-c', '1', ping_url])
+            return True
+        except:
+            return False
+
+    def _urlopen(self):
+        try:
+            urlopen(self.url, timeout=self.timeout)
+            return True
+        except:
+            return False
 
     def online_status(self):
+        icon = self.icon_off
+        color = self.color_off
+
+        if self.use_ping:
+            self.py3.log('!!!!!!!!!!!!!!!!!! use ping')
+            if self._ping():
+                self.py3.log('!!!!!!!!!!!!!!!!!! always pinging')
+                icon = self.icon_on
+                color = self.color_on
+            else:
+                self.use_ping = False
```

Comment:
None - False means don't use ping not unknown even if there is currently no difference